catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,700 @@
+                    .equals(UPDATED_TOPIC)) {
+                processUpdate(mcard);
+            } else if (event.getTopic()
+                    .equals(DELETED_TOPIC)) {
+                processDelete(mcard);
+            }
+        });
+
+    }
+
+    protected void processCreate(Metacard metacard) {
+        try {
+            updateRegistryConfigurations(metacard, true);
+        } catch (IOException | InvalidSyntaxException | ParserException e) {
+            LOGGER.error("Unable to update registry configurations, metacard still ingested");
+        }
+    }
+
+    protected void processUpdate(Metacard metacard) {
+        try {
+            updateRegistryConfigurations(metacard, false);
+        } catch (IOException | InvalidSyntaxException | ParserException e) {
+            LOGGER.error("Unable to update registry configurations, metacard still updated");
+        }
+    }
+
+    protected void processDelete(Metacard metacard) {
+        try {
+            if (cleanUpOnDelete) {
+                deleteRegistryConfigurations(metacard);
+            }
+        } catch (IOException | InvalidSyntaxException e) {
+            LOGGER.error("Unable to delete registry configurations, metacard still deleted");
+        }
+    }
+
+    /**
+     * Finds all source configurations associated with the registry metacard creates/updates them
+     * with the information in the metacards service bindings. This method will enable/disable
+     * configurations if the right conditions are met but will never delete a configuration other
+     * than for switching a configuration from enabled to disabled or vice-versa
+     * @param metacard Registry metacard with new service binding info
+     * @param createEvent Flag indicating if this was a metacard create or update event
+     * @throws IOException
+     * @throws InvalidSyntaxException
+     * @throws ParserException
+     */
+    private void updateRegistryConfigurations(Metacard metacard, boolean createEvent)
+            throws IOException, InvalidSyntaxException, ParserException {
+
+        boolean identityNode =
+                metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE) != null;
+
+        boolean autoActivateConfigurations = activateConfigurations && !identityNode && (createEvent
+                || !preserveActiveConfigurations);
+
+        List<ServiceBindingType> bindingTypes = getServiceBindings(metacard);
+
+        String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+
+        String configId = getDeconflictedConfigId(metacard.getTitle(), registryId);
+
+        Map<String, Configuration> fpidToConfigurationMap = getCurrentConfigurations(registryId);
+
+        String bindingTypeToActivate = getBindingTypeToActivate(bindingTypes);
+        String fPidToActivate = bindingTypeToFactoryPidMap.get(bindingTypeToActivate);
+
+        if (autoActivateConfigurations) {
```

Comment:
Any way we could extract this `if` statement and the `for` loop in separate `private` methods? If this business logic of manipulating the `Configuration` objects and dealing with `ServiceBindingTypes` is something that's core to the registry and can be done some other ways (endpoint, UI, etc.), and not only when receiving events, may be worth considering abstracting that out into one or more common classes.